Patch file for making QF's version of frikbot part of QF's version
of the quakeworld progs.

Index: Makefile
===================================================================
RCS file: /project/cvs/game-source/quakeworld/Makefile,v
retrieving revision 1.3
diff -u -r1.3 Makefile
--- Makefile	2 Mar 2003 04:06:46 -0000	1.3
+++ Makefile	25 Aug 2003 01:17:32 -0000
@@ -1,5 +1,7 @@
 all: qwprogs.dat
-qwprogs.dat: progs.src *.qc
-	qfcc -Werror -Wall -g
+qwprogs.dat: qwprogs.o
+	qfcc -g -o qwprogs.dat qwprogs.o -lfrikbot -lr
+qwprogs.o: progs.src *.qc
+	qfcc -Dself=@self -g -c -o qwprogs.o
 clean:
-	rm -f core *.dat *.sym progdefs.h
+	rm -f core *.dat *.sym progdefs.h *.o
Index: client.qc
===================================================================
RCS file: /project/cvs/game-source/quakeworld/client.qc,v
retrieving revision 1.7
diff -u -r1.7 client.qc
--- client.qc	4 Mar 2003 23:15:54 -0000	1.7
+++ client.qc	25 Aug 2003 01:17:33 -0000
@@ -969,6 +969,8 @@
 void ()
 PlayerPostThink =
 {
+	BotImpulses ();
+
 //	dprint ("post think\n");
 	if (self.view_ofs == '0 0 0')
 		return;         // intermission or finale
@@ -1004,6 +1006,8 @@
 void ()
 ClientConnect =
 {
+	ClientInRankings ();
+
 	bprint (PRINT_HIGH, self.netname);
 	bprint (PRINT_HIGH, " entered the game\n");
 
@@ -1022,6 +1026,8 @@
 void ()
 ClientDisconnect =
 {
+	ClientDisconnected ();
+
 	// let everyone else know
 	bprint (PRINT_HIGH, self.netname);
 	bprint (PRINT_HIGH, " left the game with ");
Index: defs.qc
===================================================================
RCS file: /project/cvs/game-source/quakeworld/defs.qc,v
retrieving revision 1.4
diff -u -r1.4 defs.qc
--- defs.qc	2 Mar 2003 04:06:47 -0000	1.4
+++ defs.qc	25 Aug 2003 01:17:33 -0000
@@ -1,7 +1,8 @@
+#include "frikbot.h"
 // SOURCE FOR GLOBALVARS_T C STRUCTURE ========================================
 
 // system globals
-entity          self;
+//entity          self;
 entity          other;
 entity          world;
 float           time;
@@ -518,7 +519,7 @@
 void(entity e, vector min, vector max) setsize = #4;
 //void() break                                            = #6;
 float() random                                          = #7;           // returns 0 - 1
-void(entity e, float chan, string samp, float vol, float atten) sound = #8;
+//void(entity e, float chan, string samp, float vol, float atten) sound = #8;
 vector(vector v) normalize                      = #9;
 void(string e) error                            = #10;
 void(string e) objerror                         = #11;
@@ -557,7 +558,7 @@
 float(entity e) checkbottom                     = #40;          // true if self is on ground
 float(vector v) pointcontents           = #41;          // returns a CONTENT_*
 float(float f) fabs = #43;
-vector(entity e, float speed) aim = #44;                // returns the shooting vector
+//vector(entity e, float speed) aim = #44;                // returns the shooting vector
 float(string s) cvar = #45;                                             // return cvar.value
 void(string s) localcmd = #46;                                  // put string into local que
 entity(entity e) nextent = #47;                                 // for looping through all ents
@@ -593,7 +594,7 @@
 string(string s) precache_sound2        = #76;          // registered version only
 string(string s) precache_file2         = #77;          // registered version only
 
-void(entity e) setspawnparms            = #78;          // set parm1... to the
+//void(entity e) setspawnparms            = #78;          // set parm1... to the
 												// values at level start
 												// for coop respawn
 void(entity killer, entity killee) logfrag = #79;       // add to stats
Index: spectate.qc
===================================================================
RCS file: /project/cvs/game-source/quakeworld/spectate.qc,v
retrieving revision 1.3
diff -u -r1.3 spectate.qc
--- spectate.qc	4 Mar 2003 23:15:54 -0000	1.3
+++ spectate.qc	25 Aug 2003 01:17:33 -0000
@@ -19,6 +19,8 @@
 void ()
 SpectatorConnect =
 {
+	ClientFixRankings ();
+
 	bprint (PRINT_MEDIUM, "Spectator ");
 	bprint (PRINT_MEDIUM, self.netname);
 	bprint (PRINT_MEDIUM, " entered the game\n");
Index: world.qc
===================================================================
RCS file: /project/cvs/game-source/quakeworld/world.qc,v
retrieving revision 1.5
diff -u -r1.5 world.qc
--- world.qc	4 Mar 2003 23:15:54 -0000	1.5
+++ world.qc	25 Aug 2003 01:17:33 -0000
@@ -17,6 +17,8 @@
 void ()
 worldspawn =
 {
+	BotInit ();
+
 	lastspawn = world;
 	InitBodyQueue ();
 
@@ -189,6 +191,8 @@
 void ()
 StartFrame =
 {
+	BotFrame ();
+
 	timelimit = cvar ("timelimit") * 60;
 	fraglimit = cvar ("fraglimit");
 	teamplay = cvar ("teamplay");
